src/rax.c

```diff
@@ -1471,27 +1485,16 @@ int raxIteratorPrevStep(raxIterator *it, int noup) {
 
         /* Try visiting the prev child if there is at least one
          * child. */
-        if (!it->node->iscompr && it->node->size > (old_noup ? 0 : 1)) {
-            raxNode **cp = raxNodeLastChildPtr(it->node);
-            int i = it->node->size-1;
-            while (i >= 0) {
-                debugf("SCAN PREV %c\n", it->node->data[i]);
-                if (it->node->data[i] < prevchild) break;
-                i--;
-                cp--;
-            }
-            /* If we found a new subtree to explore in this node,
-             * go deeper following all the last children in order to
-             * find the key lexicographically greater. */
-            if (i != -1) {
-                debugf("SCAN found a new node\n");
-                /* Enter the node we just found. */
-                if (!raxIteratorAddChars(it,it->node->data+i,1)) return 0;
-                if (!raxStackPush(&it->stack,it->node)) return 0;
-                memcpy(&it->node,cp,sizeof(it->node));
-                /* Seek sub-tree max. */
-                if (!raxSeekGreatest(it)) return 0;
-            }
+        if (!it->node->iscompr && it->node->size > (old_noup ? 0 : 1) && it->child_offset != 0) {
+            it->child_offset -= 1;
```

Comment:
I think it should be `it->child_offset -= (noup ? 0 : 1);`, when `noup` is true, it should continue to find the current node, not the previous node.
Not sure if I'm right, because I tested and found that the test failed.

Reply:
```suggestion
        if (!it->node->iscompr && it->child_offset >= (noup ? 0 : 1)) {
            it->child_offset -= (noup ? 0 : 1);
```

Reply:
Ohh, Sorry, I just realized I wrote it wrong, it should not be `noup`, but `old_noup`.

Reply:
i think we should expand on rax-tests, #10152 has already introduced the test, so maybe we can take it first and add a new test.

Reply:
Actually using `old_noup` failed tests. It looks like we can simplify more, by removing `noup/old_noup` - I've done that and it passes tests.

Reply:
Yes, that's what I've been wondering for the last two days, when trying to get the prev node of this node, we shouldn't even go to its children, because its children are always bigger than it.

Reply:
Hmm - the children of prev node are always bigger than prev node and smaller than current node. So the subtree under prev node could contain a key that is smaller than current key, I think.

Reply:
I finally figured out why I was confused.
Following example:
```c
rax *r = raxNew();
raxInsert(r,(unsigned char*)"A",1,(void*)1,NULL);
raxInsert(r,(unsigned char*)"AB",2,(void*)2,NULL);
raxInsert(r,(unsigned char*)"AC",2,(void*)3,NULL);
raxInsert(r,(unsigned char*)"AD",2,(void*)4,NULL);
raxShow(r);

# output
[A] -> [BCD]=0x1
        `-(B) []=0x2
        `-(C) []=0x3
        `-(D) []=0x4

raxSeek(&ri,"<","AE",2);
# When after call `raxLowWalk()`, `it->child_offset` will be 3(after char `D`).
# I'm not sure if using `child_offset` in this way will cause confusion,
# I always thought it was in range [0, node->size-1].
```